File: epaperless_fe/src/util/MiliSearch.js
export const miliPagination = async () => {
  try {
    const response = await fetch(
      "http://localhost:7700/indexes/epaperless/search",
      {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          offset: 0,
          limit: 4,
        }),
      }
    );

    if (!response.ok) {
      throw new Error(`HTTP error ${response.status}`);
    }

    return await response.json();
  } catch (error) {
    return error;
  }
};

miliPagination();

File: ePaperless_be/util/mili_search.py
import requests
import os


from dotenv import load_dotenv
load_dotenv()
MILI_SEARCH_BASE_URL = os.getenv("MILI_SEARCH_BASE_URL")
MILI_SEARCH_INDEX = os.getenv("MILI_SEARCH_INDEX")



def search_with_pagination(limit=10, offset=0):
    
    endpoint = f"{MILI_SEARCH_BASE_URL}/indexes/{MILI_SEARCH_INDEX}/search"
    headers = {"Content-Type": "application/json"}
    data = { "offset": offset, "limit": limit}

    response = requests.post(endpoint, headers=headers, json=data)
    result = response.json()
    
    return result['hits']

